src/codegen/constants.rs

```diff
@@ -0,0 +1,61 @@
+use analysis::imports::Imports;
+use env::Env;
+use std::path::Path;
+use file_saver;
+use codegen::general::{self, cfg_condition, cfg_deprecated, version_condition,
+                       version_condition_string};
+use library;
+
+pub fn generate(env: &Env, root_path: &Path, mod_rs: &mut Vec<String>) {
+    let path = root_path.join("constants.rs");
+    let mut imports = Imports::new(&env.library);
+
+    let has_bindable_constants = env.analysis.constants.
+        iter().
+        any(|c| *env.type_(c.typ) == library::Type::Fundamental(library::Fundamental::Utf8));
+
+    if !has_bindable_constants {
```

Comment:
Don't better check type in analysis phase and only check length here?

Reply:
I think you're right, updated!